config/clusters/qcl/common.values.yaml

```diff
@@ -54,3 +54,142 @@ jupyterhub:
+                  mem_guarantee: 28.0G
+                  cpu_guarantee: 0.4
+        kubespawner_override:
+          cpu_limit: null
+          mem_limit: null
+          node_selector:
+            node.kubernetes.io/instance-type: n2-highmem-4
+      - display_name: "Medium: up to 16 CPU / 128 GB RAM"
+        description: *profile_list_description
+        slug: medium
+        profile_options:
+          requests:
+            # NOTE: Node share choices are in active development, see comment
+            #       next to profileList: above.
+            display_name: Node share
+            choices:
+              mem_1:
+                display_name: ~1 GB, ~0.125 CPU
+                kubespawner_override:
+                  mem_guarantee: 0.969G
+                  cpu_guarantee: 0.013
+              mem_2:
+                display_name: ~2 GB, ~0.25 CPU
+                kubespawner_override:
+                  mem_guarantee: 1.938G
+                  cpu_guarantee: 0.025
+              mem_4:
+                default: true
+                display_name: ~4 GB, ~0.5 CPU
+                kubespawner_override:
+                  mem_guarantee: 3.875G
+                  cpu_guarantee: 0.05
+              mem_8:
+                display_name: ~8 GB, ~1.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 7.75G
+                  cpu_guarantee: 0.1
+              mem_16:
+                display_name: ~16 GB, ~2.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 15.5G
+                  cpu_guarantee: 0.2
+              mem_32:
+                display_name: ~32 GB, ~4.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 31.0G
+                  cpu_guarantee: 0.4
+              mem_64:
+                display_name: ~64 GB, ~8.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 62.0G
+                  cpu_guarantee: 0.8
+              mem_128:
+                display_name: ~128 GB, ~16.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 124.0G
+                  cpu_guarantee: 1.6
+        kubespawner_override:
+          cpu_limit: null
+          mem_limit: null
+          node_selector:
+            node.kubernetes.io/instance-type: n2-highmem-16
+      - display_name: "n2-highcpu-32: 32 CPU / 32 GB RAM"
+        description: "Start a container on a dedicated node"
+        slug: "n2_highcpu_32"
+        kubespawner_override:
+          node.kubernetes.io/instance-type: n2-highcpu-32
+          mem_guarantee: 28.887G
+          cpu_guarantee: 3.2
+      # TODO(pnasrat): check on value once node running
+      - display_name: "n2-highcpu-96: 96 CPU / 96 GB RAM"
+        description: "Start a container on a dedicated node"
+        slug: "n2_highcpu_96"
+        kubespawner_override:
+          node.kubernetes.io/instance-type: n2-highcpu-96
+          mem_guarantee: 86.662G
+          cpu_guarantee: 9.6
```

Comment:
Ah you are consistent with it the requests here on nodes that users isn't sharing also, sure!

Since we specify a nodeSelector to spawn on a certain node, and we wish to provide users with dedicated nodes (1:1 pod:node), we can make any request as long as one user, but not two, fits on the node.